### PR TITLE
Nimbus: runtime log level selection and logfile option

### PR DESCRIPTION
**Pull request by stefantalpalaru**
(and some drive-by fixes and improvements in some argument-related code)

### CODE REVIEW

File: nimbus/nim.cfg [hunk 1]
```diff
@@ -1,5 +1,6 @@
 -d:chronicles_line_numbers
--d:"chronicles_sinks=textblocks"
+-d:"chronicles_sinks=textblocks[file]"
```
Comment (zah) [inline nimbus/nim.cfg]: I like the `textblocks` format, but the team decided that we'll be using `textlines` by default.

File: nimbus.nimble [hunk 2]
```diff
@@ -36,5 +36,5 @@ task test, "Run tests":
   test "test_rpc"
 
 task nimbus, "Build Nimbus":
-  buildBinary "nimbus", "nimbus/"
+  buildBinary "nimbus", "nimbus/", "-d:chronicles_log_level=TRACE"
```
Comment (zah) [inline nimbus.nimble]: The idea we agreed upon was that `trace` can be used so often that it creates performance issues (even with filtering enabled). Most of the time it should be selectively activated it in a particular topic/module you are working on. Perhaps we don't have enough tracing to be a problem yet, but `DEBUG` might be a better default for a release build.

Comment (stefantalpalaru) [reply nimbus.nimble]: It's not a performance issue right now. We can revisit this when it becomes a measurable problem or when we produce a release build and decide on the compile time flags for it.

File: nimbus/nim.cfg (hunk 1 above)
Comment (stefantalpalaru) [reply nimbus/nim.cfg]: Changed.

**Pull request merged**